.readthedocs.yaml

```diff
@@ -10,6 +10,9 @@ build:
   os: ubuntu-22.04
   tools:
     python: "3.11"
+  jobs:
+    post_checkout:
+      - make .sphinx/requirements.txt
```

Comment:
This will probably fail when the docs are in a `docs` subfolder - which is a setup we support.

I would avoid using `make` here, because this introduces yet another level of ambiguity - some teams leave the starter pack Makefile in their `docs` folder and don't change it but run it from there. But `make docs/.sphinx/requirements.txt` would then fail since it would use the Makefile from the root folder.
Other teams incorporate the content from the Makefile into their own Makefile in the root folder, which would make `cd docs; make .sphinx/requirements.txt` fail.

So I suggest running the script directly here, without going through `make`. We'll still need different commands (with or without `cd docs`) depending on the directory setup, but that is the same kind of change we need for line 27/30 below. This is documented [here](https://github.com/canonical/sphinx-docs-starter-pack/blob/main/readme.rst#documentation-in-a-code-repository) and implemented [here](https://github.com/canonical/sphinx-docs-starter-pack/blob/585ab49af4764c0aeb429b04c377416049d6480a/init.sh#L20) - so those two places need updates as well.

Reply:
Note: I do not know how other projects with a `docs/` folder are set up, but the packaging guide just forwards all invocations to the `docs/Makefile` see:

https://github.com/canonical/ubuntu-packaging-guide/blob/5284d17b3852e3adff15e90081802e3ffdab98e9/Makefile#L1-L5

Running `make .sphinx/requirements.txt` works therefore totally fine. 

```
$ make .sphinx/requirements.txt
make -C docs .sphinx/requirements.txt
make[1]: Entering directory '/home/dviererbe/Repositories/ubuntu-packaging-guide/docs'
make[1]: '.sphinx/requirements.txt' is up to date.
make[1]: Leaving directory '/home/dviererbe/Repositories/ubuntu-packaging-guide/docs'
```

Question-1: Is this Makefile not present in all downstream that use a `docs/` directory?
Question-2: If this is not the case, should we add this to the default setup? Because this seems to me more stupid and simple and improves the user experience when invoking make from the root of the project.

Reply:
I feel this is outside of the scope of the starter pack. 

When a team integrates the starter pack into their code repo, we shouldn't do anything to "their" Makefile but leave it up to them how they want to work with it. 
The starter pack works out of the box if you run all commands within your docs folder. You can include the targets into the main Makefile, either by copying them over or by forwarding the invocations as you suggest. But this might not work for all teams - for example, a lot of teams probably have a `make install` already. Or they might not be using `make` for their main project.

Also, RTD doesn't use the Makefile at all, so it's "only" used for the local build, which means it doesn't have the highest priority for documentation, and in theory, teams could just get rid of it and rely on RTD only.

So adding a dependency on the Makefile into `.readthedocs.yaml` isn't a good idea imo.

Reply:
Note: Agree, thanks for the context!

Reply:
Note: 
- I removed make with https://github.com/canonical/sphinx-docs-starter-pack/pull/174/commits/69442f87b81f5f4cee4cbe1d46f77666f10e059f
- Added the path replacement to the `init.sh` script with https://github.com/canonical/sphinx-docs-starter-pack/pull/174/commits/48ffbfac8a4981bd7ce697ea625c6598c41f5b21
- Updated the extensions configurations notes in the readme with https://github.com/canonical/sphinx-docs-starter-pack/pull/174/commits/81c292e29ad5e5a5110a5229d9db91f71603b789

Question: Have I missed anything in the readme and/or do we need to document the changes somewhere else?

Reply:
> Question: Have I missed anything in the readme and/or do we need to document the changes somewhere else?

The readme looks good to me (except for a minor formatting issue).
We'll need to add this change to the [change log](https://github.com/canonical/sphinx-docs-starter-pack/wiki/Change-log) when it's merged so users know this is a breaking change.